incubator/hnc/internal/reconcilers/object.go

```diff
@@ -431,23 +432,9 @@ func (r *ObjectReconciler) syncSource(ctx context.Context, log logr.Logger, src
 	ns := r.Forest.Get(src.GetNamespace())
 	ns.SetSourceObject(src.DeepCopy())
 
-	if !r.shouldPropagateSource(log, src) {
-		// If an object was *previously* propagated by HNC, it will already be in the forest, and all
-		// the propagated copies will think they can stick around too. If, for some reason, we _now_
-		// want to stop propagating it (e.g. because we've changed the sync mode in the HNCConfig
-		// object), we need to delete it from the forest and enqueue all its propagated copies. The
-		// propagated copies will then see that that the source is missing from the forest, and delete
-		// themselves.
-		r.syncUnpropagatedSource(ctx, log, src)
-		return
-	}
-
-	// Signal the config reconciler for reconciliation because it is possible that a source object is
-	// added to the apiserver.
-	hnccrSingleton.requestReconcile("possible new source object")
-
-	// Enqueue all the descendant copies
-	r.enqueueDescendants(ctx, log, src, "possible new or modified source object")
+	// Enqueue propagated copies for this possibly deleted source
+	r.syncUnpropagatedSource(ctx, log, src)
+	return
```

Comment:
If you combined the "syncUnpropagatedSource" and the "syncPropagatedSource" (old line 445-450) maybe update the `syncUnpropagatedSource(arg..)` name to `syncPropagation(arg.., bool)` with a new boolean to indicate whether to propagate or delete since the enqueue message is different. Or change the enqueue message to more general without saying `deleted source` or `new or modified source`. 

Overall, simplifying the logic lgtm. Just need a little cleanup on the `syncUnpropagatedSource()` func.

Reply:
+1 to all of this, especially making the enqueue message more general.

Reply:
done